packages/calcite-components/src/components/modal/modal.tsx

```diff
@@ -426,16 +437,6 @@ export class Modal extends LitElement implements OpenCloseComponent {
 
     this.titleId = ensureId(this.titleEl);
     this.contentId = ensureId(this.contentEl);
-
-    if (getNearestOverflowAncestor(this.el) === document.body) {
```

Comment:
This adds back the dependency on the internal `embedded` property, which was removed by https://github.com/Esri/calcite-design-system/pull/9545. I think this is fine since it's internal. 

That said, we could possibly avoid the prop if we can confirm this behavior should always be enabled whenever an overlay component:

1. has no scroller container ancestor and 
2. `preventDocumentScroll` + `opened` are both true.

Reply:
Can we get a follow up issue for this one? I think it would be nice to get rid of the prop if possible.

Reply:
For sure. Forgot to mention that I'll be creating one. 😅

Reply:
☝️ https://github.com/Esri/calcite-design-system/issues/12011